docs/cli-arguments.md

```diff
@@ -25,34 +25,34 @@ spec:
-      --pod-namespace string             Name of the namespace of the pod specified by --pod. When set, it is expected that --pod and --pod-namespace are both set. Most likely this should be passed via the downward API. This is used for auto-detecting sharding. If set, this has preference over statically configured sharding. This is experimental, it may be removed without notice.
-      --port int                         Port to expose metrics on. (default 8080)
-      --resources string                 Comma-separated list of Resources to be enabled. Defaults to "certificatesigningrequests,configmaps,cronjobs,daemonsets,deployments,endpoints,horizontalpodautoscalers,ingresses,jobs,leases,limitranges,mutatingwebhookconfigurations,namespaces,networkpolicies,nodes,persistentvolumeclaims,persistentvolumes,poddisruptionbudgets,pods,replicasets,replicationcontrollers,resourcequotas,secrets,services,statefulsets,storageclasses,validatingwebhookconfigurations,volumeattachments"
-      --shard int32                      The instances shard nominal (zero indexed) within the total number of shards. (default 0)
-      --skip_headers                     If true, avoid header prefixes in the log messages
-      --skip_log_headers                 If true, avoid headers when opening log files
-      --stderrthreshold severity         logs at or above this threshold go to stderr (default 2)
-      --telemetry-host string            Host to expose kube-state-metrics self metrics on. (default "0.0.0.0")
-      --telemetry-port int               Port to expose kube-state-metrics self metrics on. (default 8081)
-      --total-shards int                 The total number of shards. Sharding is disabled when total shards is set to 1. (default 1)
-  -v, --v Level                          number for the log level verbosity
-      --version                          kube-state-metrics build version information
-      --vmodule moduleSpec               comma-separated list of pattern=N settings for file-filtered logging
+      --add_dir_header                    If true, adds the file directory to the header of the log messages
+      --alsologtostderr                   log to standard error as well as files
+      --apiserver string                  The URL of the apiserver to use as a master
+      --enable-gzip-encoding              Gzip responses when requested by clients via 'Accept-Encoding: gzip' header.
+  -h, --help                              Print Help text
+      --host string                       Host to expose metrics on. (default "0.0.0.0")
+      --kubeconfig string                 Absolute path to the kubeconfig file
+      --labels-metric-allow-list string   By default all labels metric will only have resource name and namespace labels, to allow more labels pass in a list of resource names in their plural form and Kubernetes labels you would like to allow, =namespaces=[k8s-label1,k8s-labeln...],pods[app]...
```

Comment:
Its hard to explain the `labels-metric-allow-list` flag due to the nature of the metric labels and k8s labels, if anyone has a more clear naming, open to suggestions! I have a similar idea for annotations where it would be --annotations-metric-allow-list and you pass the resource and annotations to allow.

Reply:
I would suggest to change the order and explain first what the flag does, then its default. Maybe something like: `Allows to pass a list of additional Kubernetes label keys that will be used in the resource' labels metric. By default the metric contains only name and namespace labels. To include additional labels provide a list of resource names in their plural form and Kubernetes label keys you would like to allow for them`